ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -253,9 +253,9 @@ public void start() throws StoreException {
         index = new PersistentIndex(dataDir, storeId, taskScheduler, log, config, factory, recovery, hardDelete,
             diskIOScheduler, metrics, time, sessionId, storeDescriptor.getIncarnationId());
         compactor.initialize(index);
-        if (config.storeRebuildTokenBasedOnCompactionHistory
+        if (config.storeRebuildTokenBasedOnCompactionHistory || (replicaId != null
```

Comment:
QQ: This means we could enable this feature on an entire host or selected partitions?